itest/lnd_route_blinding.go

```diff
@@ -0,0 +1,290 @@
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/lightningnetwork/lnd/chainreg"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/routerrpc"
+	"github.com/lightningnetwork/lnd/lntest"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// testQueryBlindedRoutes tests querying routes to blinded routes. To do this,
+// it sets up a nework of Alice - Bob - Carol and creates a mock blinded route
+// that uses Carol as the introduction node (plus dummy hops to cover multiple
+// hops). The test simply asserts that the structure of the route is as
+// expected.
+func testQueryBlindedRoutes(ht *lntest.HarnessTest) {
+	var (
+		ctxb = context.Background()
+
+		// Convenience aliases.
+		alice = ht.Alice
+		bob   = ht.Bob
+	)
+
+	// Setup a two hop channel network: Alice -- Bob -- Carol.
+	// We set our proportional fee for these channels to zero, so that
+	// our calculations are easier. This is okay, because we're not testing
+	// the basic mechanics of pathfinding in this test.
+	chanAmt := btcutil.Amount(100000)
+	chanPointAliceBob := ht.OpenChannel(
+		alice, bob, lntest.OpenChannelParams{
+			Amt:        chanAmt,
+			BaseFee:    10000,
+			FeeRate:    0,
+			UseBaseFee: true,
+			UseFeeRate: true,
+		},
+	)
+
+	carol := ht.NewNode("Carol", nil)
+	ht.EnsureConnected(bob, carol)
+
+	var bobCarolBase uint64 = 2000
+	chanPointBobCarol := ht.OpenChannel(
+		bob, carol, lntest.OpenChannelParams{
+			Amt:        chanAmt,
+			BaseFee:    bobCarolBase,
+			FeeRate:    0,
+			UseBaseFee: true,
+			UseFeeRate: true,
+		},
+	)
+
+	// Wait for Alice to see Bob/Carol's channel because she'll need it for
+	// pathfinding.
+	ht.AssertTopologyChannelOpen(alice, chanPointBobCarol)
+
+	// Lookup full channel info so that we have channel ids for our route.
+	aliceBobChan := ht.GetChannelByChanPoint(alice, chanPointAliceBob)
+	bobCarolChan := ht.GetChannelByChanPoint(bob, chanPointBobCarol)
+
+	// Sanity check that bob's fee is as expected.
+	chanInfoReq := &lnrpc.ChanInfoRequest{
+		ChanId: bobCarolChan.ChanId,
+	}
+
+	bobCarolInfo, err := bob.RPC.LN.GetChanInfo(ctxb, chanInfoReq)
+	require.NoError(ht, err)
+
+	// Our test relies on knowing the fee rate for bob - carol to set the
+	// fees we expect for our route. Perform a quick sanity check that our
+	// policy is as expected.
+	var policy *lnrpc.RoutingPolicy
+	if bobCarolInfo.Node1Pub == bob.PubKeyStr {
+		policy = bobCarolInfo.Node1Policy
+	} else {
+		policy = bobCarolInfo.Node2Policy
+	}
+	require.Equal(ht, bobCarolBase, uint64(policy.FeeBaseMsat), "base fee")
+	require.EqualValues(ht, 0, policy.FeeRateMilliMsat, "fee rate")
+
+	// We'll also need the current block height to calculate our locktimes.
+	ctxt, cancel := context.WithTimeout(ctxb, defaultTimeout)
+	defer cancel()
+
+	info, err := alice.RPC.LN.GetInfo(ctxt, &lnrpc.GetInfoRequest{})
```

Comment:
we now have a new set of methods, so can just all `alice.RPC.GetInfo` here

Reply:
<img width="452" alt="Screenshot 2023-08-24 at 9 32 44 AM" src="https://github.com/lightningnetwork/lnd/assets/42311294/7516c256-3cec-4b55-8a69-047ca069e752">